api/types/databaseobjectimportrule/databaseobjectimportrule.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Gravitational, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package databaseobjectimportrule
+
+import (
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/defaults"
+	dbobjectimportrulev1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/dbobjectimportrule/v1"
+	headerv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/header/v1"
+	"github.com/gravitational/teleport/api/types"
+)
+
+// NewDatabaseObjectImportRule creates a new dbobjectimportrulev1.DatabaseObjectImportRule.
+func NewDatabaseObjectImportRule(name string, spec *dbobjectimportrulev1.DatabaseObjectImportRuleSpec) (*dbobjectimportrulev1.DatabaseObjectImportRule, error) {
+	return NewDatabaseObjectImportRuleWithLabels(name, nil, spec)
+}
+
+// NewDatabaseObjectImportRuleWithLabels creates a new dbobjectimportrulev1.DatabaseObjectImportRule with specified labels.
+func NewDatabaseObjectImportRuleWithLabels(name string, labels map[string]string, spec *dbobjectimportrulev1.DatabaseObjectImportRuleSpec) (*dbobjectimportrulev1.DatabaseObjectImportRule, error) {
+	out := &dbobjectimportrulev1.DatabaseObjectImportRule{
+		Kind:    types.KindDatabaseObjectImportRule,
+		Version: types.V1,
+		Metadata: &headerv1.Metadata{
+			Name:      name,
+			Namespace: defaults.Namespace,
+			Labels:    labels,
+		},
+		Spec: spec,
+	}
+
+	err := ValidateDatabaseObjectImportRule(out)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return out, nil
+}
+
+// ValidateDatabaseObjectImportRule checks if dbobjectimportrulev1.DatabaseObjectImportRule is valid.
+func ValidateDatabaseObjectImportRule(rule *dbobjectimportrulev1.DatabaseObjectImportRule) error {
+	if rule == nil {
+		return trace.BadParameter("must be non-nil")
```

Comment:
Ditto for ambiguous error.

```suggestion
		return trace.BadParameter("database object import rule be non-nil")
```

Reply:
You still have the stack trace, but it doesn't cost much to improve this, so done.

Reply:
I've changed the message.